src/pages/uxp/reference-js/Modules/uxp/Persistent File Storage/FileSystem.md

```diff
@@ -0,0 +1,396 @@
+| oldPath | `string` | path where the old file name to change is located |
+| newPath | `string` | path where the new file name will be |
+| callback | `Array<n` | if not provided, this function will return Promise object |
+
+**Example**  
+```js
+fs.rename("plugin-data:/oldName.txt", "plugin-temp:/newName.txt");
+```
+
+
+<a name="module-storage-filesystem-copyfile" id="module-storage-filesystem-copyfile"></a>
+
+## copyFile(srcPath, destPath, flags, callback)
+Copies a file or a folder from the source path to the destination path
+
+**Returns**: `Promise<int>` - 0 if succeeded, otherwise throws an Error  
+
+| Param | Type | Default | Description |
+| --- | --- | --- | --- |
+| srcPath | `string` |  | path where the source file to copy is located |
+| destPath | `string` |  | path where the source file will be copied to |
+| flags | `int` | `0` | see flags in [uv_fs_copyfile](https://docs.libuv.org/en/v1.x/fs.html) |
+| callback | `function` |  | if not provided, this function will return Promise object |
+
+**Example**  
+```js
+const data = fs.copyFile("plugin-data:/copyFrom.txt", "plugin-temp:/copyTo.txt");
+```
+
+
+<a name="module-storage-filesystem-unlink" id="module-storage-filesystem-unlink"></a>
+
+## unlink(path, callback)
+Deletes a name with the file it refers to asynchronously
+
+**Returns**: `Promise<int>` - 0 if succeeded, otherwise throws an Error  
+
+| Param | Type | Description |
+| --- | --- | --- |
+| path | `string` | path where the file to delete is located |
+| callback | `function` | if not provided, this function will return Promise object |
+
+**Example**  
+```js
+await fs.unlink("plugin-data:/fileToDelete.txt");
+```
+
+
+<a name="module-storage-filesystem-mkdir" id="module-storage-filesystem-mkdir"></a>
+
+## mkdir(path, callback)
+Creates a directory of the path asynchronously
+
+**Returns**: `Promise<int>` - 0 if succeeded, otherwise throws an Error  
+
+| Param | Type | Description |
+| --- | --- | --- |
+| path | `string` | path where to create the directory |
+| callback | `function` | if not provided, this function will return Promise object |
+
+**Example**  
+```js
+await fs.mkdir("plugin-data:/newDir");
+```
+
+
+<a name="module-storage-filesystem-rmdir" id="module-storage-filesystem-rmdir"></a>
+
+## rmdir(path, callback)
+Removes a directory asynchronously
+
+**Returns**: `Promise<int>` - 0 if succeeded, otherwise throws an Error  
+
+| Param | Type | Description |
+| --- | --- | --- |
+| path | `string` | path where to remove the directory |
+| callback | `function` | if not provided, this function will return Promise object |
+
+**Example**  
+```js
+await fs.rmdir("plugin-data:/dirToRemove");
+```
+
+
+<a name="module-storage-filesystem-readdir" id="module-storage-filesystem-readdir"></a>
+
+## readdir(path, callback)
+Reads a directory to list the containing files and directories asynchronously
+
+**Returns**: `Promise<strings>` - string array of containing files and directories in the path  
```

Comment:
```suggestion
**Returns**: `Promise<string[]>` - string array of containing files and directories in the path  
```

Reply:
Possibly postponed to a "data type normalization" effort, but given that it's unclear as it is, I'll leave this one unresolved for now.

Reply:
🔴